python-package/lightgbm/engine.py

```diff
@@ -199,6 +199,8 @@ def train(params, train_set, num_boost_round=100,
         callbacks = set()
     else:
         for i, cb in enumerate(callbacks):
+            if getattr(cb, 'first_metric_only', False) and feval is not None:
+                raise LightGBMError("`first_metric_only` and `feval` are not available at the same time.")
```

Comment:
@matsuken92 Sorry, forgot to ask in my previous comment. Do we really need this limitation after cpp changes?